deltacat/compute/compactor/compaction_session.py

```diff
@@ -343,188 +347,194 @@ def _execute_compaction_round(
                 records_per_primary_key_index_file,
                 delete_prev_primary_key_index,
             )
+    # memray official documentation link: https://bloomberg.github.io/memray/getting_started.html
+    with memray.Tracker(
+            f"compaction_session.bin"
+    ) if enable_profiler else nullcontext():
```

Comment:
We could add this in the `compact_partition` to reduce the overall cyclomatic complexity of this method.

Reply:
Don e